includes/Node.h

```diff
@@ -1,29 +1,73 @@
 #ifndef NODE_H
 #define NODE_H
 
-template<typename DataType>
+#include <vector>
+
+// define point with logitude, latitude, 2 points make a Rect for bounding box
+class Point {
+public:
+    Point();
+    ~Point();
+
+    Point(double, double); // longtitude, latitude
+    Point(double, double, int);
+
+    bool operator==(const Point&) const;
+    bool operator!=(const Point&) const;
+
+    const double& getLong() const { return longtitude; }
+    const double& getLat() const { return latitude; }
+
+private:    
+    double longtitude, latitude;
+    int id;
+};
+
+// define a bounding box with 2 points, left lower and right upper
+class Rect {
+public:
+    Rect();
+    ~Rect();
+
+    Rect(Point, Point, int);
+
+    bool operator==(const Rect&) const;
+    bool operator!=(const Rect&) const;
+
+    const Point& getLower() const { return lower; }
+    const Point& getUpper() const { return upper; } 
+
+private:
+    Point lower;
```

Comment:
Do you mean lower-left and upper-right?  Usually we name them as `ll` and `ur`, respectively.

And since you keep them private, it always saves you maintenance time by following the convention to prefix `m_` in the data members.

Reply:
Done, thank you!